ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/ReadSymbolsFromMobileStyle/ReadSymbolsFromMobileStyle.qml

```diff
@@ -0,0 +1,339 @@
+            }
+        }
+
+        onSearchSymbolsStatusChanged: {
+            if (searchSymbolsStatus !== Enums.TaskStatusCompleted)
+                return;
+
+            // get initial list of symbols
+            searchSymbolsResult.forEach(function(result){
+                if (result.category === "Face") {
+                    currentFace = result;
+                }
+            });
+
+            // fetch a new symbol with the initial symbol layer keys
+            updateSymbol();
+        }
+
+        onFetchSymbolStatusChanged: {
+            if (fetchSymbolStatus !== Enums.TaskStatusCompleted)
+                return;
+
+            // set the current symbol
+            currentSymbol = fetchSymbolResult;
+
+            // set the color locked preferences per layer
+            currentSymbol.symbolLayers.forEach(function(symbolLyr) {
+                symbolLyr.colorLocked = true;
+            });
+            currentSymbol.symbolLayers.get(0).colorLocked = false;
+
+            // update size
+            currentSymbol.size = sizeSlider.value;
+
+            // update color
+            currentSymbol.color = colorComboBox.currentText;
+
+            // update swatch
+            currentSymbol.swatchImageChanged.connect(function(){
+                symbolImage.source = currentSymbol.swatchImage;
+            });
+            currentSymbol.createSwatch();
+        }
+    }
+
+    function updateSymbol() {
+        if (!currentFace || !currentEyes || !currentMouth || !currentHat)
+            return;
+
+        symbolStyle.fetchSymbolWithKeyList([currentFace.key, currentEyes.key, currentMouth.key, currentHat.key]);
+    }
+
+    Rectangle {
+        anchors {
+            fill: optionGrid
+            margins: -5
+        }
+        color: "#efefef"
+    }
+
+    GridLayout {
+        id: optionGrid
+        anchors {
+            left: parent.left
+            top: parent.top
+            margins: 10
+        }
+
+        columns: 2
+
+        Label {
+            Layout.columnSpan: 2
+            Layout.alignment: Qt.AlignCenter
+            text: "Change Options to Modify Symbol"
+            font.underline: true
+        }
+
+        Label {
+            text: "Eyes"
+        }
+
+        ComboBox {
+            id: eyeComboBox
+            property var style
+            model: style ? style.searchSymbolsResult : []
+            textRole: "name"
+            onCurrentTextChanged: {
+                currentEyes = style.searchSymbolsResult.get(currentIndex);
+                updateSymbol();
+            }
+            Component.onCompleted: {
+                // Create a SymbolStyle and only fetch the Eye symbols
+                style = ArcGISRuntimeEnvironment.createObject("SymbolStyle", {
+                                                                  styleLocation: dataPath
+                                                              },
+                                                              eyeComboBox);
```

Comment:
You're setting the `eyeComboBox` as the parent of the style? Is that needed? Same question applies below.

Reply:
it was in my original code, but then i made style a property of the combo box, so that should hold a reference to it and keep it from getting gc'd. ill remove it